src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -34,7 +34,7 @@ internal partial class Binder
             // Create a map from type parameter name to ordinal.
             // No need to report duplicate names since duplicates
             // are reported when the type parameters are bound.
-            var names = new Dictionary<string, int>(n);
+            var names = new Dictionary<string, int>(n, StringOrdinalComparer.Instance);
```

Comment:
Oh, gotcha. Instance is the name of the singleton member, not "check equality by checking to see if they are same instance".

In which case, the main difference is that this is using the FNV hashing algorithm, which is one of the fastest hashing functions.